bugbug/models/component.py

```diff
@@ -203,3 +204,42 @@ def is_meaningful(product, component):
 
     def get_feature_names(self):
         return self.extraction_pipeline.named_steps["union"].get_feature_names()
+
+    def check(self):
+        success = super().check()
+
+        # Get the number of bugs per full component to fasten up the check
+        bugs_number = get_product_component_count()
+
+        # Check that the most meaningful product components stills have at
+        # least a bug in this component. If the check is failing that could
+        # means that:
+        # - A component has been renamed / removed
+        # - TODO: Complete this list
+
+        for product, component in self.meaningful_product_components:
+            full_comp = f"{product}::{component}"
+
+            if full_comp not in bugs_number.keys():
+                msg = f"Component {component!r} of product {product!r} have 0 bugs in it, failure"
+                print(msg)
+                success = False
+
+        # Check that the conflated components still exists and have at least
+        # one bug
+
+        # Assert all conflated components are either in conflated_components_mapping or exist as components.
```

Comment:
This is actually only checking that the conflated components match some real components

Reply:
I've added the check `conflated_component in self.CONFLATED_COMPONENTS_MAPPING`, I think that's what was missing.

I'm not sure why we need to check if it is matching real components or if it's part of the mapping. It it right that a conflated component is in the mapping without matching any real component?

Reply:
What we want to check:
1) product/components in meaningful_product_components still exist and have more than 0 bugs
2) conflated components in CONFLATED_COMPONENTS match at least one component which has more than 0 bugs
3) values of CONFLATED_COMPONENTS_MAPPING still exist as components and have more than 0 bugs
4) conflated components in CONFLATED_COMPONENTS either exist as components or are in CONFLATED_COMPONENTS_MAPPING
5) there is no component with many bugs that is not in meaningful_product_components

4 should check if the items of CONFLATED_COMPONENTS are either full components or they are in CONFLATED_COMPONENTS_MAPPING, not if they just match some real component (which is checked in 2).